ethportal-api/build.rs

```diff
@@ -1,36 +1,15 @@
-use std::{fs::File, io::Write};
-
-use shadow_rs::SdResult;
-
-fn main() -> SdResult<()> {
-    shadow_rs::new_hook(hook)?;
-
-    Ok(())
-}
-
-fn hook(mut file: &File) -> SdResult<()> {
-    let env_var_git_hash = std::env::var("GIT_HASH").unwrap_or_default();
-    writeln!(file, "const ENV_GIT_HASH: &str = \"{}\";", env_var_git_hash)?;
-
-    hook_method(file)?;
-
-    Ok(())
-}
-
-fn hook_method(mut file: &File) -> SdResult<()> {
-    let hook_fn = r#"
-pub const fn short_commit() -> &'static str {
-    if shadow_rs::str_get!(SHORT_COMMIT, 0).is_some() {
-        return SHORT_COMMIT;
-    }
-
-    if shadow_rs::str_get!(ENV_GIT_HASH, 0).is_some() {
-        ENV_GIT_HASH
-    } else {
-        "unknown"
-    }
-}"#;
-
-    writeln!(file, "{}", hook_fn)?;
+use std::error::Error;
+
+use vergen::EmitBuilder;
+
+fn main() -> Result<(), Box<dyn Error>> {
```

Comment:
I think you're going to either have to set a `GIT_HASH` env var here, or update our 2 dockerfiles so that it'll pick up the env var [here](https://github.com/ethereum/trin/blob/master/docker/Dockerfile#L11)

Reply:
https://github.com/sigp/lighthouse/blob/stable/Dockerfile
https://github.com/paradigmxyz/reth/blob/main/Dockerfile

both Lighthouse and Reth copy the whole folder, if we do that it would resolve this issue and we could remove

https://github.com/ethereum/trin/blob/ddbd9ff54df221c2ccf6967fab8340553896b723/docker/Dockerfile#L8-L11

and passing the git hash in with the CI, 

As only copying in the crates would be problematic for vergen

Reth uses this https://github.com/LukeMathWalker/cargo-chef?tab=readme-ov-file#benefits-of-cargo-chef package, which I think will resolve our caching problems we were trying to solve by copying crates individually, this should also make it so we have to update our crates less so the code is more maintainable.

Reply:
I will make a PR updating our dockerfiles with what I said above in mind

Reply:
https://github.com/ethereum/trin/pull/1619

Here is a PR which resolves your concern, once both of these PR's are merged it will work as expected